### DISCUSSION TITLE
Need Apple MIDI Arduino Advice

### DISCUSSION THREAD

**Discussion by jhaimerl1** [lathoub/Arduino-AppleMIDI-Library#151] [Show and tell]
![Slide1](https://user-images.githubusercontent.com/112823025/189736406-2df41b49-5ed3-4384-8967-ace2428aeea6.jpeg)

I wanted to ask the Arduino Apple MIDI community  if my planned project makes sense and is possible...

Background: 
_For my band, I run Ableton Live on my Macbook. For each song we play, I run a Live session that has tracks that generate MIDI commands for my equipment and controls our light shows.  USB MIDI is used for my Helix Guitar processor and Boss VE500 vocal processor control. I also have a network "Session 1"  MIDI for our iPAD displays. A track sends the iPads song selection and scrolling commands to the iPads, which use Songbook+ to display PDFs.  All this works great._

_I'm adding the Arduino to control the Soundcraft Mixer recorder.  I have written a few test that allow me to read status and control the recording enable.  I have built a wired MIDI interface to the Arduino and successfully sent commands from my MacBook to control the recorder.  I want to convert it to Apple MIDI because I would also like to have the Arduino Controller keep track of a few items and be able to send some midi commands to the ipads (status alerts)._

**If the Arduino and 5 iPads are on the Apple MIDI "Session 1" on the Macbook, I know the MacBook can send/receive MIDI from the Arduino... and that the MacBook can send MIDI to the iPads... Question is can I have the Arduino also send MIDI to the iPads?????  I haven't been able to get this to work yet.**

Is there a trick in the MIDI Network Setup?

Thanks for any advice/guidance.

JH

**Comment by lathoub** [comment]
I don't recommend using WiFi, as the [latency is considerable](https://github.com/lathoub/Arduino-AppleMIDI-Library/wiki/Keeping-Latency-under-control), consider [BLE](https://github.com/lathoub/Arduino-BLE-MIDI).

I refer to others for the network topology

**Reply by jhaimerl1** [reply]
OK. Yes. All the time critical Helix and VE500 commands are over USB. These switch fast enough that I successfully change settings between beats/notes :)  

My Macbook today runs Apple MIDI Network Session 1 today to 5 iPads wirelessly with no latency issues. MIDI commands from Ableton on the Macbook call up the correct PDF file in Songbook+ and then scrolls them at the correct time. This is not time critical and works great.

I would like to add the Arduino with a LAN cable to be a MIDI controller for the Soundcraft. I plan to have Ableton on Macbook send a MIDI command (wireless to router, wired to Arduino) to start/stop the data recorder with a LAN interface.  From what I've experimented with today, this looks possible... (Although I haven't tested a LAN connection from Arduino to Soundcraft Mixer while simultaneously having another Apple MIDI connection to MacBook... if not possible, I can always connect it to a 1/4" TS connector on the mixer).

What I haven't been able to do is on this same Apple MIDI Network Session1 have the Arduino send MIDI and have the iPads receive it.  I'd like to have it bring up a warning by MIDI command between songs... I was assuming that when I had a session on the Macbook with multiple devices, that the devices could ALL communicate with each other, not just the Macbook.  Again, I haven't been able to do this yet...  Is there something special I need to do to be able to have remote devices on the session communicate with each other?

Thank You

**Reply by jhaimerl1** [reply]
I was experimenting and created 2 sessions:
Session 1: connected iPads
Session 2: connected Arduino
Then in ableton I created a track that takes Session 1 as an input and outputs to Session 2.  I had to restart things and then it seemed to work. I was able to have the iPads see the notes that the Arduino was sending.

I would prefer to have everything on Session 1, as described in my last post, if that is possible (?)
